src/lib/utils/synapseTypes/Entity.ts

```diff
@@ -4,19 +4,34 @@ import { ConcreteEntityType } from './ConcreteEntityType'
-  name: string // The name of this entity.
-  description?: string // The description of this entity.
-  id?: string // The unique immutable ID for this entity. A new ID will be generated for new Entities. Once issued, this ID is guaranteed to never change or be re-issued
-  uri?: string // This field is deprecated and will be removed in future versions of Synapse
-  etag?: string // Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent updates. Since the E-Tag changes every time an entity is updated it is used to detect when a client's current representation of an entity is out-of-date.
-  createdOn?: string // The date this entity was created.
-  modifiedOn?: string // The date this entity was last modified.
-  createdBy?: string // The ID of the user that created this entity.
-  modifiedBy?: string // The ID of the user that last modified this entity.
-  annotations?: string // This field is deprecated and will be removed in future versions of Synapse
-  accessControlList?: string // This field is deprecated and will be removed in future versions of Synapse
-  parentId?: string // The ID of the Entity that is the parent of this Entity.
-  entityType?: string // This field is deprecated and will be removed in future versions of Synapse
-  concreteType: ConcreteEntityType // Indicates which implementation of Entity this object represents. It should be set to one of the following: org.sagebionetworks.repo.model.Project, org.sagebionetworks.repo.model.Folder, or org.sagebionetworks.repo.model.FileEntity.
+  /** The name of this entity? */
+  name: string
+  /** The description of this entity. */
+  description?: string
+  /** The unique immutable ID for this entity. A new ID will be generated for new Entities. Once issued, this ID is guaranteed to never change or be re-issued */
+  readonly id?: string
+  /** Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent updates. Since the E-Tag changes every time an entity is updated it is used to detect when a client's current representation of an entity is out-of-date. */
+  readonly etag?: string
+  /** The date this entity was created. */
+  readonly createdOn?: string
+  /** The date this entity was last modified. */
+  readonly modifiedOn?: string
+  /** The ID of the user that created this entity. */
+  readonly createdBy?: string
+  /** The ID of the user that last modified this entity. */
+  readonly modifiedBy?: string
+  /** The ID of the Entity that is the parent of this Entity. */
+  parentId?: string
+  /** Indicates which implementation of Entity this object represents. */
+  readonly concreteType: ConcreteEntityType
+  /** @deprecated This field is deprecated and will be removed in future versions of Synapse */
+  attachments?: AttachmentData[]
+  /** @deprecated This field is deprecated and will be removed in future versions of Synapse */
```

Comment:
We should use this comment style when possible because VSCode (hopefully other IDEs too?) displays/hints this info inline:

![image](https://user-images.githubusercontent.com/17580037/120669857-d9340600-c45d-11eb-879f-47834d13be7b.png)

Reply:
Like this idea.  Any way to help devs work towards this style (other than this comment)?  Something in VSCode configuration?  Generating ts definitions from the schema would certainly do the trick :)